go/libkb/home_nix_test.go

```diff
@@ -67,18 +69,34 @@ func TestDarwinHomeFinderInDev(t *testing.T) {
 }
 
 func TestPosixRuntimeDir(t *testing.T) {
-	var home string
+	var cmdHome string
 	env := make(map[string]string)
 	ge := func(s string) string { return env[s] }
-	hf := NewHomeFinder("tester", func() string { return home }, nil, nil, "posix", func() RunMode { return ProductionRunMode }, makeLogGetter(t), ge)
+	hf := NewHomeFinder("tester", func() string { return cmdHome }, nil, nil, "posix", func() RunMode { return ProductionRunMode }, makeLogGetter(t), ge)
 
-	// Test the case that the --home flag overrides XDG_RUNTIME_DIR
-	home = "/footown"
+	origHomeEnv := os.Getenv("HOME")
+
+	// Custom env, custom cmd, XDG set
+	cmdHome = "/footown"
+	env["HOME"] = "/yoyo"
+	env["XDG_RUNTIME_DIR"] = "/barland"
+	require.Equal(t, "/footown/.config/tester", hf.RuntimeDir(), "expect custom cmd to win")
+
+	// Custom env, no cmd, XDG set
+	cmdHome = ""
```

Comment:
This is the behavior I changed. If you set a home variable that is different than what your home is according to /etc/passwd, it uses that home variable.